### PR TITLE
Add more Span Batch unit tests

### LINKED ISSUE #20: [Feature] Span Batch Tests
### Component

protocol

### Describe the feature you would like

A few corner cases of `SpanBatch` are not sufficiently covered by unit tests.

- Block timestamp less than L1 origin
- Missing l1 origin for empty batch
- Overlapping block tx count mismatch
- Overlapping block tx doesn't match

The above should be tested as part of this issue, alongside any other added test coverage.

### Provenance

https://github.com/anton-rs/kona/issues/681

### PR DESCRIPTION
**Pull request by sergerad**
Add more unit test for `SpanBatch`, namely:
- Batch block timestamp less than L1 origin
- Overlapping block tx count mismatch
- Overlapping block tx content match
- Missing L1 origin for singular batch

Closes #20

### CODE REVIEW

**Comment by sergerad** [comment]
@emhane I have added the tests you suggested in the issue. LMK if you think of any more. When I get a chance to go deeper I can also try think of some others.

TY

File: crates/protocol/src/batch/span.rs [hunk 1]
```diff
@@ -108,31 +108,36 @@ impl SpanBatch {
...
-            let single_batch = SingleBatch {
-                epoch_num: batch.epoch_num,
-                epoch_hash: origin_epoch_hash,
-                timestamp: batch.timestamp,
-                transactions: batch.transactions.clone(),
-                ..Default::default()
-            };
-            single_batches.push(single_batch);
-        }
-        Ok(single_batches)
```
Comment (sergerad) [inline crates/protocol/src/batch/span.rs:135]: This change is not necessary. Could also move the `mut origin_index` into the try_fold but would be at expense of readability (singles vec turns into a tuple).

Happy to revert this change if you prefer the previous impl.

Comment (refcell) [reply crates/protocol/src/batch/span.rs:135]: I'd prefer to keep the previous format only because it's a bit more digestible and flatter nesting structure (less indentations that is).

Comment (refcell) [reply crates/protocol/src/batch/span.rs:135]: This likely won't need to change so shouldn't cause to much of a headache as long as we document well.

**Review by refcell — COMMENTED** [review]
These tests are very nice! If we can revert the functional change, gtg. Great job

File: crates/protocol/src/batch/span.rs (hunk 1 above)
Comment (sergerad) [reply crates/protocol/src/batch/span.rs]: Reverted, thanks!

**Review by emhane — APPROVED** [review]
pending @refcell or @clabby
